### PR TITLE
Fix inheritance of data models

### PR DESCRIPTION
**Pull request by nden**
Resolves [JP-nnnn](https://jira.stsci.edu/browse/JP-nnnn)

Closes #

This PR addresses an issue with data model inheritance. Models inheriting from `stdatamodels.jwst.DataModel` do not register as instances of `DataModel`. For example `isinstance(SlitModel(), stdatamodels.jwst.datamodels.DataModel` returns `False`. Since `stdatamodels.jwst.datamodels.DataModel` is deprecated n favor of `stdatamodels.jwst.datamodels.JwstDataModel` I've replaced any occurrence of `DatModel` with `JwstDataModel`. 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

regression tests: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/695/

### CODE REVIEW

File: jwst/outlier_detection/outlier_detection_ifu.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __init__(self, input_models, reffiles=None, **pars):
         drizzled_models : list of objects
             ModelContainer containing drizzled grouped input images
 
-        reffiles : dict of ~jwst.datamodels.DataModel
+        reffiles : dict of `~stdatamodels.jwst.datamodels.jwstDataModel`
```
Comment (hbushouse) [inline jwst/outlier_detection/outlier_detection_ifu.py:59]: Need a capital "J" in `jwstDataModel`

**Review by hbushouse — COMMENTED** [review]
Just caught one minor typo.

**Comment by hbushouse** [comment]
When you do a regtest run, be sure to specify "stdatamodels==1.4.0" in the OVERRIDE_REQUIREMENTS field, so that you don't get spurious errors/differences due to the new NIRCam grism transforms.

**Review by hbushouse — APPROVED** [review]
LGTM

**Comment by hbushouse** [comment]
Is there a github issue or JP ticket tracking this problem?

**Comment by nden** [comment]
No, there isn't a ticket. Should I write a Github or JP ticket?

**Comment by tapastro** [comment]
I think there are many more occurrences of `DataModel` in the pipeline - should they all be changed to `JwstDataModel`, or were these changes targeted?

**Comment by hbushouse** [comment]
> No, there isn't a ticket. Should I write a Github or JP ticket?

I think a github issue would be good and sufficient. No need to bother folks who monitor the JP project.

**Comment by nden** [comment]
> I think there are many more occurrences of `DataModel` in the pipeline - should they all be changed to `JwstDataModel`, or were these changes targeted?

Are there? In code or docstrings? I guess I missed them. I'd like to fix all of them.

**Pull request merged**